### PR TITLE
LinkageMonitor to find latest BOM version

### LINKED ISSUE #720: LinkageMonitor: to pick up latest BOM
LinkageMonitor to pick up latest BOM from Maven Central.

### PR DESCRIPTION
**Pull request by suztomo**
Fixes #720

### CODE REVIEW

File: linkage-monitor/src/main/java/com/google/cloud/tools/dependencies/linkagemonitor/LinkageMonitor.java [hunk 1]
```diff
@@ -109,17 +118,36 @@ static Bom copyWithSnapshot(RepositorySystem repositorySystem, Bom bom)
   static Optional<String> findSnapshotVersion(
       RepositorySystem repositorySystem, RepositorySystemSession session, Artifact artifact)
       throws VersionRangeResolutionException {
+    String version = findHighestVersion(repositorySystem, session, artifact);
```
Comment (elharo) [inline linkage-monitor/src/main/java/com/google/cloud/tools/dependencies/linkagemonitor/LinkageMonitor.java]: Does this load from the local repo too? That's the only place snapshot versions are likely to be right now.

File: linkage-monitor/src/main/java/com/google/cloud/tools/dependencies/linkagemonitor/LinkageMonitor.java [hunk 2]
```diff
@@ -109,17 +118,36 @@ static Bom copyWithSnapshot(RepositorySystem repositorySystem, Bom bom)
...
-      return Optional.of(version);
-    }
-    return Optional.empty();
+    return versionResult.getHighestVersion().toString();
+  }
+
+  private String findLatestCoordinates(String groupId, String artifactId)
+      throws VersionRangeResolutionException {
+    RepositorySystemSession session = RepositoryUtility.newSession(repositorySystem);
+    Artifact artifact = new DefaultArtifact(groupId, artifactId, null,
```
Comment (elharo) [inline linkage-monitor/src/main/java/com/google/cloud/tools/dependencies/linkagemonitor/LinkageMonitor.java]: why not just start with a version range here?

File: linkage-monitor/src/main/java/com/google/cloud/tools/dependencies/linkagemonitor/LinkageMonitor.java [hunk 3]
```diff
@@ -47,18 +51,23 @@
...
+    new LinkageMonitor().run(coordinatesElements.get(0), coordinatesElements.get(1));
   }
 
-  private void run(String bomCoordinates)
+  private void run(String groupId, String artifactId)
       throws RepositoryException, IOException, LinkageMonitorException {
-    Bom baseline = RepositoryUtility.readBom(bomCoordinates);
+    String latestBomCoordinates = findLatestCoordinates(groupId, artifactId);
+    System.out.println("BOM Coordinates: " + latestBomCoordinates);
+    Bom baseline = RepositoryUtility.readBom(latestBomCoordinates);
```
Comment (elharo) [inline linkage-monitor/src/main/java/com/google/cloud/tools/dependencies/linkagemonitor/LinkageMonitor.java]: Possibly I should have kept the method that reads from an Artifact. We're doing a lot of back and forth between artifacts and BOM coordinates.

File: linkage-monitor/src/main/java/com/google/cloud/tools/dependencies/linkagemonitor/LinkageMonitor.java [hunk 4]
```diff
@@ -109,17 +118,36 @@ static Bom copyWithSnapshot(RepositorySystem repositorySystem, Bom bom)
   static Optional<String> findSnapshotVersion(
```
Comment (elharo) [inline linkage-monitor/src/main/java/com/google/cloud/tools/dependencies/linkagemonitor/LinkageMonitor.java]: where else is this used? Can it be private or @VisibleForTesting?

File: linkage-monitor/src/main/java/com/google/cloud/tools/dependencies/linkagemonitor/LinkageMonitor.java [hunk 5]
```diff
@@ -82,8 +92,26 @@ private void run(String bomCoordinates)
...
+  }
+
+  /**
+   * Validates the difference of linkage erros between {@code snapshotSymbolProblems} and {@code
+   * baselineSymbolProblems}.
+   *
+   * @throws {@link LinkageMonitorException} if there are new linkage errors
+   */
+  @VisibleForTesting
+  static void validateDifference(
```
Comment (suztomo) [inline linkage-monitor/src/main/java/com/google/cloud/tools/dependencies/linkagemonitor/LinkageMonitor.java]: Splitting this function makes it easy to write test for the logic.

**Pull request merged**